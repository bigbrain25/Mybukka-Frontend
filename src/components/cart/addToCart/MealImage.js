import React from 'react';
import './MealImage.scss';
import { useCloudinayService } from '../../img/Cloudinary';

const MealImage = ({ imageUrl, toggleAddToCart }) => {
  const { domain, supports } = useCloudinayService();
  let ext = 'jpg', queryString = '';
  const lqh = 120;// low quality height
  const options = {
    h: 640, // height
    w: 'auto', // width
    q: 85, // quality
    c: 'scale' // mode
  };

  const [storageClienId, imgSrc] = imageUrl.replace(domain, '').split('upload');
  const imgSrcWithoutExt = imgSrc.replace(/\.(jpe?g|gif|png|PNG|svg|webp)$/, '');

  // Loop through option prop and build queryString
  Object.keys(options).map((option, i) => queryString += `${i < 1 ? '' : ','}${option}_${options[option]}`); // eslint-disable-line

  const lqipQueryString = queryString.replace(`h_${options.h}`, `h_${lqh}`);

  // If a format has not been specified, detect webp support
  if (supports.webp) {
    ext = 'webp';
  }

  return (
    <div id="" className="Meal-Image">
      <div>
        <img alt={imgSrcWithoutExt.split('/')[1]} src={`${domain}${storageClienId}upload/${lqipQueryString}${imgSrcWithoutExt}.${ext}`} className="MealImage-Invisible" />
        <img alt={imgSrcWithoutExt.split('/')[1]} src={`${domain}${storageClienId}upload/${queryString}${imgSrcWithoutExt}.${ext}`} className="MealImage-Invisible" />
        <div title={imgSrcWithoutExt.split('/')[1]} className="MealImage-Visible-Filter" style={{ backgroundImage: `url(${domain}${storageClienId}upload/${lqipQueryString}${imgSrcWithoutExt}.${ext})`, opacity: 1 }} />
        <div title={imgSrcWithoutExt.split('/')[1]} className="MealImage-Visible" style={{ backgroundImage: `url(${domain}${storageClienId}upload/${queryString}${imgSrcWithoutExt}.${ext})`, opacity: 1 }} />
      </div>
      <div className="MealImage-Resize" />
      <div
        className="MealImage-Left-Icon"
        role="button"
        arai-pressed="false"
        tabIndex="0"
        onClick={() => toggleAddToCart(false)}
      >
        <svg width="24" height="24" fill="currentColor" viewBox="0 0 24 24">
          <defs><path id="icon-close_svg__a" d="M0 1.5L1.5 0 8 6.5 14.5 0 16 1.5 9.5 8l6.5 6.5-1.5 1.5L8 9.5 1.5 16 0 14.5 6.5 8z" /></defs>
          <use xlinkHref="#icon-close_svg__a" transform="translate(4 4)" />
        </svg>
      </div>
    </div>
  );
};
export default MealImage;
