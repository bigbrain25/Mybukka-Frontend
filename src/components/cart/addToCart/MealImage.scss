.Meal-Image {
  position: relative;
  background-color: transparent;
  height: auto;
  width: 100%;
  flex-shrink: 0;
  overflow: hidden;
  @media screen and (min-width: 1060px) {
    height: 100%;
    max-height: 600px;
    max-width: 600px;
  }
}

.Meal-Image::after {
  @media screen and (max-width: 1059px) {
    content: "";
    position: absolute;
    top: 0px;
    left: 0px;
    right: 0px;
    bottom: 0px;
    width: 100%;
    height: 100%;
    z-index: inherit;
    background-color: rgba(45, 49, 56, 0.03);
  }
}

.MealImage-Invisible {
  display: none;
}

.MealImage-Visible-Filter {
  position: absolute;
  background-size: cover;
  top: 0px;
  left: 0px;
  filter: blur(50px);
  transform: scale(1);
  height: 100%;
  width: 100%;
  background-position: center center;
  background-repeat: no-repeat;
  transition: opacity 0.4s linear 0s;
  @media screen and (min-width: 1060px) {
    box-shadow: rgba(45, 49, 56, 0.08) -15px 0px 10px -10px inset;
  }
}

.MealImage-Visible {
  position: absolute;
  background-size: cover;
  top: 0px;
  left: 0px;
  height: 100%;
  width: 100%;
  background-position: center center;
  background-repeat: no-repeat;
  transition: opacity 0.4s linear 0s;
  @media screen and (min-width: 1060px) {
    box-shadow: rgba(45, 49, 56, 0.08) -15px 0px 10px -10px inset;
  }
}

.MealImage-Resize {
  height: auto;
  padding-bottom: 100%
}

.MealImage-Left-Icon {
  position: absolute;
  left: 16px;
  top: 16px;
  z-index: 100000;
  color: rgb(255, 255, 255);
  cursor: pointer;
  @media screen and (min-width: 768px) {
    left: 24px;
    top: 24px;
  }
  @media screen and (min-width: 1060px) {
    display: none;
    left: 36px;
    top: 36px;
  }
}
