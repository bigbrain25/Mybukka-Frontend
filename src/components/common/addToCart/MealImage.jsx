import React from 'react';

import PropTypes from 'prop-types';
import toggleAddToCartModal from 'Redux/toggleAddToCartModal';

import generateImageSizes from 'Utilities/generateScreenSizeImageUrl';
import Column from 'Components/grid/Column';
import Cancel from 'Components/icons/Cancel';

import './mealImage.scss';

const MealImage = ({ imageUrl, toggleAddToCart }) => (
  <Column classNames="col-12 col-xs-12 col-sm-12 col-md-12 col-lg-7 pl-0 img-column">
    <div
      role="button"
      arai-pressed="false"
      tabIndex="0"
      className="cancel-modal-meal d-lg-none obvious-bg"
      onClick={() => toggleAddToCart(false)}
    >
      <Cancel />
    </div>
    <img src={generateImageSizes(imageUrl, ['650', 600])} className="image-details img-fluid" alt="" />
  </Column>
);

export default MealImage;

MealImage.propTypes = {
  imageUrl: PropTypes.string.isRequired
};
