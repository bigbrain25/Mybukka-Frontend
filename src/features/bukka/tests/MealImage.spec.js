import React from 'react';
import MealImage from '../addToCart/MealImage';

describe('Meal Image Component', () => {
  const wrapper = mount(<MealImage imageUrl="http://some-img-url" />);

  it('renders properly', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
