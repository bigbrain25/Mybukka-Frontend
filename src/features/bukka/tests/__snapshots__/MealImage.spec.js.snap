// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Meal Image Component renders properly 1`] = `
ReactWrapper {
  Symbol(enzyme.__unrendered__): <MealImage
    imageUrl="http://some-img-url"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "imageUrl": "http://some-img-url",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <div
            className="cancel-modal-meal d-lg-none"
            data-dismiss="modal"
          >
            <Cancel />
          </div>,
          <img
            alt=""
            className="image-details img-fluid"
            src="http://some-img-url"
          />,
        ],
        "classNames": "col-12 col-xs-12 col-sm-12 col-md-12 col-lg-7 img-column",
      },
      "ref": null,
      "rendered": Object {
        "instance": <div
          class="col col-12 col-xs-12 col-sm-12 col-md-12 col-lg-7 img-column"
        >
          <div
            class="cancel-modal-meal d-lg-none"
            data-dismiss="modal"
          >
            <svg
              fill="currentColor"
              height="24"
              viewBox="0 0 24 24"
              width="24"
            >
              <defs>
                <path
                  d="M0 1.5L1.5 0 8 6.5 14.5 0 16 1.5 9.5 8l6.5 6.5-1.5 1.5L8 9.5 1.5 16 0 14.5 6.5 8z"
                  id="icon-close_svg__a"
                />
              </defs>
              <use
                transform="translate(4 4)"
                xlink:href="#icon-close_svg__a"
              />
            </svg>
          </div>
          <img
            alt=""
            class="image-details img-fluid"
            src="http://some-img-url"
          />
        </div>,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="cancel-modal-meal d-lg-none"
              data-dismiss="modal"
            >
              <Cancel />
            </div>,
            <img
              alt=""
              className="image-details img-fluid"
              src="http://some-img-url"
            />,
          ],
          "className": "col col-12 col-xs-12 col-sm-12 col-md-12 col-lg-7 img-column",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": <div
              class="cancel-modal-meal d-lg-none"
              data-dismiss="modal"
            >
              <svg
                fill="currentColor"
                height="24"
                viewBox="0 0 24 24"
                width="24"
              >
                <defs>
                  <path
                    d="M0 1.5L1.5 0 8 6.5 14.5 0 16 1.5 9.5 8l6.5 6.5-1.5 1.5L8 9.5 1.5 16 0 14.5 6.5 8z"
                    id="icon-close_svg__a"
                  />
                </defs>
                <use
                  transform="translate(4 4)"
                  xlink:href="#icon-close_svg__a"
                />
              </svg>
            </div>,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <Cancel />,
              "className": "cancel-modal-meal d-lg-none",
              "data-dismiss": "modal",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {},
                "ref": null,
                "rendered": Object {
                  "instance": <svg
                    fill="currentColor"
                    height="24"
                    viewBox="0 0 24 24"
                    width="24"
                  >
                    <defs>
                      <path
                        d="M0 1.5L1.5 0 8 6.5 14.5 0 16 1.5 9.5 8l6.5 6.5-1.5 1.5L8 9.5 1.5 16 0 14.5 6.5 8z"
                        id="icon-close_svg__a"
                      />
                    </defs>
                    <use
                      transform="translate(4 4)"
                      xlink:href="#icon-close_svg__a"
                    />
                  </svg>,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      <defs>
                        <path
                          d="M0 1.5L1.5 0 8 6.5 14.5 0 16 1.5 9.5 8l6.5 6.5-1.5 1.5L8 9.5 1.5 16 0 14.5 6.5 8z"
                          id="icon-close_svg__a"
                        />
                      </defs>,
                      <use
                        transform="translate(4 4)"
                        xlinkHref="#icon-close_svg__a"
                      />,
                    ],
                    "fill": "currentColor",
                    "height": "24",
                    "viewBox": "0 0 24 24",
                    "width": "24",
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": <defs>
                        <path
                          d="M0 1.5L1.5 0 8 6.5 14.5 0 16 1.5 9.5 8l6.5 6.5-1.5 1.5L8 9.5 1.5 16 0 14.5 6.5 8z"
                          id="icon-close_svg__a"
                        />
                      </defs>,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": <path
                          d="M0 1.5L1.5 0 8 6.5 14.5 0 16 1.5 9.5 8l6.5 6.5-1.5 1.5L8 9.5 1.5 16 0 14.5 6.5 8z"
                          id="icon-close_svg__a"
                        />,
                      },
                      "ref": null,
                      "rendered": Array [
                        Object {
                          "instance": <path
                            d="M0 1.5L1.5 0 8 6.5 14.5 0 16 1.5 9.5 8l6.5 6.5-1.5 1.5L8 9.5 1.5 16 0 14.5 6.5 8z"
                            id="icon-close_svg__a"
                          />,
                          "key": undefined,
                          "nodeType": "host",
                          "props": Object {
                            "d": "M0 1.5L1.5 0 8 6.5 14.5 0 16 1.5 9.5 8l6.5 6.5-1.5 1.5L8 9.5 1.5 16 0 14.5 6.5 8z",
                            "id": "icon-close_svg__a",
                          },
                          "ref": null,
                          "rendered": Array [
                            undefined,
                          ],
                          "type": "path",
                        },
                      ],
                      "type": "defs",
                    },
                    Object {
                      "instance": <use
                        transform="translate(4 4)"
                        xlink:href="#icon-close_svg__a"
                      />,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "transform": "translate(4 4)",
                        "xlinkHref": "#icon-close_svg__a",
                      },
                      "ref": null,
                      "rendered": Array [
                        undefined,
                      ],
                      "type": "use",
                    },
                  ],
                  "type": "svg",
                },
                "type": [Function],
              },
            ],
            "type": "div",
          },
          Object {
            "instance": <img
              alt=""
              class="image-details img-fluid"
              src="http://some-img-url"
            />,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": "",
              "className": "image-details img-fluid",
              "src": "http://some-img-url",
            },
            "ref": null,
            "rendered": Array [
              undefined,
            ],
            "type": "img",
          },
        ],
        "type": "div",
      },
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "imageUrl": "http://some-img-url",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <div
              className="cancel-modal-meal d-lg-none"
              data-dismiss="modal"
            >
              <Cancel />
            </div>,
            <img
              alt=""
              className="image-details img-fluid"
              src="http://some-img-url"
            />,
          ],
          "classNames": "col-12 col-xs-12 col-sm-12 col-md-12 col-lg-7 img-column",
        },
        "ref": null,
        "rendered": Object {
          "instance": <div
            class="col col-12 col-xs-12 col-sm-12 col-md-12 col-lg-7 img-column"
          >
            <div
              class="cancel-modal-meal d-lg-none"
              data-dismiss="modal"
            >
              <svg
                fill="currentColor"
                height="24"
                viewBox="0 0 24 24"
                width="24"
              >
                <defs>
                  <path
                    d="M0 1.5L1.5 0 8 6.5 14.5 0 16 1.5 9.5 8l6.5 6.5-1.5 1.5L8 9.5 1.5 16 0 14.5 6.5 8z"
                    id="icon-close_svg__a"
                  />
                </defs>
                <use
                  transform="translate(4 4)"
                  xlink:href="#icon-close_svg__a"
                />
              </svg>
            </div>
            <img
              alt=""
              class="image-details img-fluid"
              src="http://some-img-url"
            />
          </div>,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="cancel-modal-meal d-lg-none"
                data-dismiss="modal"
              >
                <Cancel />
              </div>,
              <img
                alt=""
                className="image-details img-fluid"
                src="http://some-img-url"
              />,
            ],
            "className": "col col-12 col-xs-12 col-sm-12 col-md-12 col-lg-7 img-column",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": <div
                class="cancel-modal-meal d-lg-none"
                data-dismiss="modal"
              >
                <svg
                  fill="currentColor"
                  height="24"
                  viewBox="0 0 24 24"
                  width="24"
                >
                  <defs>
                    <path
                      d="M0 1.5L1.5 0 8 6.5 14.5 0 16 1.5 9.5 8l6.5 6.5-1.5 1.5L8 9.5 1.5 16 0 14.5 6.5 8z"
                      id="icon-close_svg__a"
                    />
                  </defs>
                  <use
                    transform="translate(4 4)"
                    xlink:href="#icon-close_svg__a"
                  />
                </svg>
              </div>,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <Cancel />,
                "className": "cancel-modal-meal d-lg-none",
                "data-dismiss": "modal",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {},
                  "ref": null,
                  "rendered": Object {
                    "instance": <svg
                      fill="currentColor"
                      height="24"
                      viewBox="0 0 24 24"
                      width="24"
                    >
                      <defs>
                        <path
                          d="M0 1.5L1.5 0 8 6.5 14.5 0 16 1.5 9.5 8l6.5 6.5-1.5 1.5L8 9.5 1.5 16 0 14.5 6.5 8z"
                          id="icon-close_svg__a"
                        />
                      </defs>
                      <use
                        transform="translate(4 4)"
                        xlink:href="#icon-close_svg__a"
                      />
                    </svg>,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": Array [
                        <defs>
                          <path
                            d="M0 1.5L1.5 0 8 6.5 14.5 0 16 1.5 9.5 8l6.5 6.5-1.5 1.5L8 9.5 1.5 16 0 14.5 6.5 8z"
                            id="icon-close_svg__a"
                          />
                        </defs>,
                        <use
                          transform="translate(4 4)"
                          xlinkHref="#icon-close_svg__a"
                        />,
                      ],
                      "fill": "currentColor",
                      "height": "24",
                      "viewBox": "0 0 24 24",
                      "width": "24",
                    },
                    "ref": null,
                    "rendered": Array [
                      Object {
                        "instance": <defs>
                          <path
                            d="M0 1.5L1.5 0 8 6.5 14.5 0 16 1.5 9.5 8l6.5 6.5-1.5 1.5L8 9.5 1.5 16 0 14.5 6.5 8z"
                            id="icon-close_svg__a"
                          />
                        </defs>,
                        "key": undefined,
                        "nodeType": "host",
                        "props": Object {
                          "children": <path
                            d="M0 1.5L1.5 0 8 6.5 14.5 0 16 1.5 9.5 8l6.5 6.5-1.5 1.5L8 9.5 1.5 16 0 14.5 6.5 8z"
                            id="icon-close_svg__a"
                          />,
                        },
                        "ref": null,
                        "rendered": Array [
                          Object {
                            "instance": <path
                              d="M0 1.5L1.5 0 8 6.5 14.5 0 16 1.5 9.5 8l6.5 6.5-1.5 1.5L8 9.5 1.5 16 0 14.5 6.5 8z"
                              id="icon-close_svg__a"
                            />,
                            "key": undefined,
                            "nodeType": "host",
                            "props": Object {
                              "d": "M0 1.5L1.5 0 8 6.5 14.5 0 16 1.5 9.5 8l6.5 6.5-1.5 1.5L8 9.5 1.5 16 0 14.5 6.5 8z",
                              "id": "icon-close_svg__a",
                            },
                            "ref": null,
                            "rendered": Array [
                              undefined,
                            ],
                            "type": "path",
                          },
                        ],
                        "type": "defs",
                      },
                      Object {
                        "instance": <use
                          transform="translate(4 4)"
                          xlink:href="#icon-close_svg__a"
                        />,
                        "key": undefined,
                        "nodeType": "host",
                        "props": Object {
                          "transform": "translate(4 4)",
                          "xlinkHref": "#icon-close_svg__a",
                        },
                        "ref": null,
                        "rendered": Array [
                          undefined,
                        ],
                        "type": "use",
                      },
                    ],
                    "type": "svg",
                  },
                  "type": [Function],
                },
              ],
              "type": "div",
            },
            Object {
              "instance": <img
                alt=""
                class="image-details img-fluid"
                src="http://some-img-url"
              />,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "alt": "",
                "className": "image-details img-fluid",
                "src": "http://some-img-url",
              },
              "ref": null,
              "rendered": Array [
                undefined,
              ],
              "type": "img",
            },
          ],
          "type": "div",
        },
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
